lib/ansible/modules/windows/win_disk_management.ps1

```
@@ -0,0 +1,888 @@
+
+# Search disk
+$ParamsDisk = @{
+    DiskSize = $Size
+    PartitionStyle = $FindPartitionStyle
+    OperationalStatus = $OperationalStatus
+    ReadOnly = $ReadOnly
+    Number = $Number
+}
+try {
+    $disk = Search-Disk @ParamsDisk
+} catch {
+    Fail-Json -obj $result -message "Failed to search and/or select the disk with the specified option values: $($_.Exception.Message)"
+}
+if ($disk) {
+    $diskcount = $disk | Measure-Object | Select-Object  -ExpandProperty Count
+    if ($diskcount -ge 2) {
+        $disk = $disk[0]
+    }
+    [string]$DOperSt = $disk.OperationalStatus
+    [string]$DPartStyle = $disk.PartitionStyle
+    [string]$DROState = $disk.IsReadOnly
+} else {
+        Fail-Json -obj $result -message "No disk could be found and selected with the passed option values"
+}
+
+if ($check_mode -or $diff_mode) {
+    $result += @{ 
+        change_log = @{
+        }
+    }
+}
+
+# Check and set operational status and read-only state
+$SetOnline = $false
+$SetWriteable = $false
+$OPStatusFailed = $false
+$ROStatusFailed = $false
+if ($DPartStyle -ne "RAW") {
+    if ($DOperSt -ne "Online") {
+        if (-not $check_mode) {
+            # Set online
+            try {
+                Set-OperationalStatus -Disk $disk
+            } catch {
+                if ($Verbose) {
+                    $result.Remove("change_log")
+                }
+                Fail-Json -obj $result -message "Failed to set the disk online: $($_.Exception.Message)"
+            }
+            $result.changed = $true
+            $SetOnline = $true
+        }
+    }      
+    if ($DROState -eq "True") {
+        if (-not $check_mode) {
+            # Set writeable
+            try {
+                Set-DiskWriteable -Disk $disk
+            } catch {
+                if ($SetOnline) {
+                    try {
+                        Set-OperationalStatus -Disk $disk -Deactivate
+                    } catch {
+                        $OPStatusFailed = $true
```

Just have an empty catch block, changed would be whatever it is and shouldn't change regardless of this operation, i.e. you are effectively saying set changed to true when `$OpStatusFailed` as `$SetOnline` is true. `$SetOnline` is only set to true straight after you set `$result.changed = $true` a few lines above this. This behaviour is repeated a few times below as well.

You should also make the reverting actions into it's own function where you pass in `$SetOnline`, `$SetWritable` and try and revert the actions. This way you only need to call the revert function here (and below) instead of repeating yourself multiple times.